example/linux/gw-example/main.c

```diff
@@ -209,27 +252,42 @@ static bool reconnect(uint32_t timeout_s)
     size_t proto_size;
     char topic_status[sizeof(TOPIC_EVENT_PREFIX) + sizeof(m_gateway_id) + 1];
     char topic_all_requests[16 + sizeof(m_gateway_id)]; //"gw-request/+/<gateway_id/#"
-    uint8_t last_will_message[128];
     MQTTClient_connectOptions conn_opts = MQTTClient_connectOptions_initializer;
     MQTTClient_willOptions will_options = MQTTClient_willOptions_initializer;
     MQTTClient_SSLOptions ssl_options = MQTTClient_SSLOptions_initializer;
 
     conn_opts.keepAliveInterval = 20;
     conn_opts.cleansession = 1;
-    conn_opts.ssl = &ssl_options;
+    if (strlen(m_user) == 0)
```

Comment:
Why?
user/password and ssl are not related.
I didn't check but I think you can have anonymous connection with SSL

Reply:
Just related to how I think we use it (user/pwd required when using ssl, not needed if not ssl). Ok, I change that.